### PR TITLE
Added remaining documents API endpoints

### LINKED ISSUE #16: Add method update partial index
I see in the documentation there is update partial but in `@elastic/app-search-node` there is no this method.

version: `7.6.0`

https://swiftype.com/documentation/app-search/api/documents#partial

Screen Shot 2020-03-28 at 4 37 03 PM

`appSearch.js`
```js
/**
   * Partial update a batch of documents.
   *
   * @param {String} engineName unique Engine name
   * @param {Array<Object>} documents Array of document objects to be updated.
   * @returns {Promise<Array<Object>>} a Promise that returns a result {Object} when resolved, otherwise throws an Error.
   */
  updateDocuments(engineName, documents) {
    return this.client.patch(`engines/${encodeURIComponent(engineName)}/documents`, documents)
  }
```

`client.js`
```js
  patch(path, params) {
    return this._jsonRequest('PATCH', path, params)
  }
```

### PR DESCRIPTION
**Pull request by JasonStoltz**
- List Documents
- Update Documents

Closes #16

### CODE REVIEW

File: lib/appSearch.js [hunk 1]
```diff
@@ -120,12 +120,37 @@ class AppSearchClient {
...
+   */
+  updateDocuments(engineName, documents) {
+    return this.client.patch(`engines/${encodeURIComponent(engineName)}/documents`, documents)
+  }
+
+  /**
+   * List all documents
+   *
+   * @param {String} engineName unique Engine name
+   * @param {Object} options
```
Comment (richkuz) [inline lib/appSearch.js:138]: What are the supported options?

Comment (JasonStoltz) [reply lib/appSearch.js:138]: The support options are the options that the endpoint supports: https://swiftype.com/documentation/app-search/api/documents#list

Comment (JasonStoltz) [reply lib/appSearch.js]: https://github.com/elastic/app-search-node/pull/17/commits/227f19f3dcfac3055c06f305e67b175adeb4979a

**Comment by afoucret** [comment]
LGTM

**Pull request merged**